src/syscheckd/db/fim_db_files.c

```diff
@@ -330,186 +267,98 @@ char **fim_db_get_paths_from_inode(fdb_t *fim_sql, unsigned long int inode, unsi
     return paths;
 }
 
-int fim_db_insert_data(fdb_t *fim_sql, const fim_file_data *entry, int *row_id) {
-    int res;
+static int fim_db_check_limit(fdb_t *fim_sql) {
+    int nodes_count;
+    int retval = FIMDB_OK;
 
-    w_mutex_lock(&fim_sql->mutex);
+    if (syscheck.file_limit_enabled == 0) {
+        return FIMDB_OK;
+    }
 
-    if(*row_id == 0) {
-        fim_db_clean_stmt(fim_sql, FIMDB_STMT_INSERT_DATA);
+#ifndef WIN32
+    nodes_count = _fim_db_get_count(fim_sql, FIMDB_STMT_GET_COUNT_PATH);
+#else
+    nodes_count = _fim_db_get_count(fim_sql, FIMDB_STMT_COUNT_DB_ENTRIES);
+#endif
+    if (nodes_count < 0) {
+        retval = FIMDB_ERR;
+    } else if (nodes_count >= syscheck.file_limit) {
+        fim_sql->full = true;
+        retval = FIMDB_FULL;
+    }
 
-        fim_db_bind_insert_data(fim_sql, entry);
+    return retval;
+}
 
-        if (res = sqlite3_step(fim_sql->stmt[FIMDB_STMT_INSERT_DATA]), res != SQLITE_DONE) {
-            merror("Step error inserting data row_id '%d': %s", *row_id, sqlite3_errmsg(fim_sql->db));
-            w_mutex_unlock(&fim_sql->mutex);
-            return FIMDB_ERR;
-        }
+static int _fim_db_insert_entry(fdb_t *fim_sql, const char *file_path, const fim_file_data *entry) {
+    int res;
 
-        *row_id = sqlite3_last_insert_rowid(fim_sql->db);
-    } else {
-        // Update file_data
-        fim_db_clean_stmt(fim_sql, FIMDB_STMT_UPDATE_DATA);
-        fim_db_bind_update_data(fim_sql, entry, row_id);
+    fim_db_clean_stmt(fim_sql, FIMDB_STMT_REPLACE_ENTRY);
+    fim_db_bind_replace_entry(fim_sql, file_path, entry);
 
-        if (res = sqlite3_step(fim_sql->stmt[FIMDB_STMT_UPDATE_DATA]), res != SQLITE_DONE) {
-            merror("Step error updating data row_id '%d': %s", *row_id, sqlite3_errmsg(fim_sql->db));
-            w_mutex_unlock(&fim_sql->mutex);
+    if (res = sqlite3_step(fim_sql->stmt[FIMDB_STMT_REPLACE_ENTRY]), res != SQLITE_DONE) {
+            merror("Step error replacing path '%s': %s", file_path, sqlite3_errmsg(fim_sql->db));
             return FIMDB_ERR;
-        }
     }
-    w_mutex_unlock(&fim_sql->mutex);
 
     return FIMDB_OK;
 }
 
-int fim_db_insert_path(fdb_t *fim_sql, const char *file_path, const fim_file_data *entry, int inode_id) {
-    int res;
+int fim_db_insert_entry(fdb_t *fim_sql, const char *file_path, const fim_file_data *entry) {
```

Comment:
It looks like this function is no longer used anywhere, we shoule remove it if this is the case.